discovery-infra/test_infra/junit_report/junit_test_case.py

```diff
@@ -0,0 +1,62 @@
+import time
+from typing import Union, Callable, Type
+
+from junit_xml import TestCase
+
+
+class CaseFailure:
+    def __init__(self, message: str, output: str = "", ftype: str = "") -> None:
+        self._message = message
+        self._output = output
+        self._type = ftype
+
+    def __getitem__(self, item):
+        return self.__getattribute__(item)
+
+    @property
+    def message(self):
+        return self._message
+
+    @property
+    def output(self):
+        return self._output
+
+    @property
+    def type(self):
+        return self._type
+
+
+class JunitTestCase:
```

Comment:
I think this decorator deserves some documentation - what should I apply it on? Why? What does it do?